### PR TITLE
Improve documentation

### LINKED ISSUE #159: Improve documentation
The redme.md is way to log, so my solution would be to create a docs folder, and split the redme up.
With a good documentation issues like #156 would not occur.

### PR DESCRIPTION
**Pull request by benbieler**
### What's new?

The documentation has been split up, and now is in a docs folder.

Reviewers:
@sindresorhus, @tjsail33 @rajzshkr @bobthecow @ajaybc @lukeed @samarpanda

### CODE REVIEW

File: docs/gettingstarted.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+```js
+var gulp = require('gulp');
+var rev = require('gulp-rev');
+
+gulp.task('default', function () {
+	return gulp.src('src/*.css')
+		.pipe(rev())
+		.pipe(gulp.dest('dist'));
+});
+```
```
Comment (sindresorhus) [inline docs/gettingstarted.md]: I think this part should stay in the readme.

File: docs/api/rev/original_path_and_asset_hash.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+### Original path
+
+Original file paths are stored at `file.revOrigPath`. This could come in handy for things like rewriting references to the assets.
+
+
+### Asset hash
+
+The hash of each rev'd file is stored at `file.revHash`. You can use this for customizing the file renaming, or for building different manifest formats.
+
```
Comment (sindresorhus) [inline docs/api/rev/original_path_and_asset_hash.md]: This file can go in rev.md

**Comment by sindresorhus** [comment]
A bit too split up IMHO.

-

> license → licence.md

👎

File: docs/gettingstarted.md (hunk 1 above)
Comment (benbieler) [inline docs/gettingstarted.md]: @sindresorhus https://github.com/sindresorhus/gulp-rev/blob/%23159-improve-documentation/readme.md ?

File: docs/api/rev/original_path_and_asset_hash.md (hunk 2 above)
Comment (benbieler) [inline docs/api/rev/original_path_and_asset_hash.md]: Done

File: docs/gettingstarted.md (hunk 1 above)
Comment (sindresorhus) [inline docs/gettingstarted.md]: Yes, but list the different documentation documents in the `Docs` section in the readme directly.

Comment (benbieler) [inline docs/gettingstarted.md]: @sindresorhus Done

**Comment by benbieler** [comment]
@sindresorhus, can I squash and merge?

**Comment by shinnn** [comment]
> license → licence

Intentional?

**Comment by benbieler** [comment]
@shinnn yes, I personally like .md licences, but I [reverted](https://github.com/sindresorhus/gulp-rev/blob/%23159-improve-documentation/licence) this, since sindre already pointed this out.

**Comment by lukeed** [comment]
He means the typo on "licence" 😝

**Comment by sindresorhus** [comment]
@benbieler re license, it's not markdown, so no need to add extension.

**Comment by sindresorhus** [comment]
The `gettingstarted.md` file can be removed as it's just a duplicate of what's in the readme. The `docs/api/rev/rev.md` file has too much directory nesting. Just make it `docs/api.md`. Too much nesting for many of the other files too.

**Comment by benbieler** [comment]
Sorry about the license typo, I speak UK english.

File: docs/api/sourcemaps_and_gulp_concat.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+gulp.task('default', function () {
+	return gulp.src('src/*.js')
+		.pipe(sourcemaps.init())
+		.pipe(concat({path: 'bundle.js', cwd: ''}))
+		.pipe(rev())
+		.pipe(sourcemaps.write('.'))
+		.pipe(gulp.dest('dist'));
+```
+
+##### [next: Streaming] (streaming.md)
```
Comment (samarpanda) [inline docs/api/sourcemaps_and_gulp_concat.md]: @benbieler I like this next links here, it ease the navigation while go through the docs. Especially when its multiple files.

Comment (benbieler) [inline docs/api/sourcemaps_and_gulp_concat.md]: @samarpanda thanks ;)

**Comment by sindresorhus** [comment]
@benbieler The following needs to be resolved:

- https://github.com/sindresorhus/gulp-rev/pull/170#issuecomment-216019322 
- https://github.com/sindresorhus/gulp-rev/pull/170#issuecomment-216019441
- `docs/readme.md` can be removed as it's just a duplicate of the listing in the readme.

**Comment by benbieler** [comment]
@sindresorhus I already have solved those issues.(^^ See commits above). I think that an additional readme.md inside the docs folder can be quite usefull (If the user simply clicks on 'docs' ), but if you want it gone, it is gone ;)

**Comment by sindresorhus** [comment]
Not at far as I can see. look at the diff. `license` needs to be renamed back to `license.md` as asked, and less nesting per https://github.com/sindresorhus/gulp-rev/pull/170#issuecomment-216019441.

**Comment by sindresorhus** [comment]
Closing for lack of response and updates.

**Comment by benbieler** [comment]
@sindresorhus I'm sorry for the inactivity. I had a lot to do regarding work. Would this be merged if I reopened this, and fixed the small issues?

**Comment by benbieler** [comment]
I got this request from multiple people, so I think this would be worth it ;) ...

**Comment by sindresorhus** [comment]
@benbieler Nah. In hindsight, I don't think it's enough of a benefit. If the readme had been huge it would have been worth it, but it's not very long, so I think it's easier to just have everything in one file :)

**Comment by benbieler** [comment]
Ok. Fine.

**Pull request closed without merging**